src/ngraph/op/cum_sum.hpp

```diff
@@ -0,0 +1,99 @@
+       ///
+       /// |                      | Description                                        |
+       /// | -------------------- | -------------------------------------------------- |
+       /// | `reverse`            | if set to 1, performs the sum in reverse direction |
+       ///
+       /// ## Inputs
+       ///
+       /// |       | Description                                            |
+       /// | ----- | ------------------------------------------------------ |
+       /// | `arg` | An input tensor of any shape and numeric element type. |
+       ///
+       /// |       | Description                                                     |
+       /// | ----- | --------------------------------------------------------------- |
+       /// | `axis`| An axis position along which cumulative sum must be performed.  |
+       ///
+       /// ## Output
+       ///
+       /// | Description                                                                         |
+       /// | ------------------------------------------------------------------------------------|
+       /// | Output tensor of the same type as `arg` with cumulative sums of the arg's elements  |
+       //
+       // clang-format off
+       class CumSum : public Op 
+       {
+       public:
+           NGRAPH_API
+           static constexpr NodeTypeInfo type_info{ "CumSum", 0 };
+           const NodeTypeInfo& get_type_info() const override { return type_info; }
+           /// \brief Constructs a cumulative summation operation.
+           CumSum() = default;
+          
+           /// \brief Constructs a cumulative summation operation.
+           ///
+           /// \param arg The tensor to be summed.
+           /// \param axis The axis position along which cumulative sum must be performed.
+           CumSum(const Output<Node>& arg, const int64_t axis = 0, const int exclusive = 0, const int reverse = 0); 
+    
+           /// \brief Constructs a cumulative summation operation.
+           ///
+           /// \param arg The tensor to be summed.
+           /// \param axis The axis position along which cumulative sum must be performed.
+           CumSum(const Output<Node>& arg, const Output<Node>& axis, const int exclusive = 0, const int reverse = 0);
+
+           virtual std::shared_ptr<Node>
+               copy_with_new_args(const NodeVector& new_args) const override;
+
+           /// \return The default value for CumSum.
+           virtual std::shared_ptr<Node> get_default_value() const override;
+           int64_t get_axis() const { return m_axis; }
+           int is_exclusive() const { return m_exclusive; }
+           int is_reverse() const { return m_reverse; }
+       protected:
+           virtual void generate_adjoints(autodiff::Adjoints& adjoints,
+               const NodeVector& deltas) override;
+    
+       private:
+	   int m_exclusive;
+	   int m_reverse;
```

Comment:
There is a proposal to the spec which define these attributes as boolean values.
Please use `bool` instead of `int` here.

Reply:
In TF, exclusive and reverse are bools too: https://www.tensorflow.org/versions/r1.14/api_docs/cc/struct/tensorflow/ops/cumsum/attrs.html#structtensorflow_1_1ops_1_1_cumsum_1_1_attrs

Reply:
I will change them too to `bool` I initially had `bool` in mind, but onnx doc misguided me :)